### PR TITLE
:sparkles: Edit a Bot #103

### LINKED ISSUE #103: ✨  Edit a Bot
#  Edit a Bot

#### NOTE: this feature is dependent on #102, please complete that issue first.

---

Quick links: [Summary](#summary) | [Description](#description) | [Technical-Analysis](#Technical-Analysis) | [Acceptance-Criteria](#Acceptance-Criteria)

---

### Summary

As a user I want to edit the image, name and description of my bots even after they are created.

### Description

Currently, users are only able to create the bots but once they are created they can't edit any detail about their bots. To help the user achieve this we'll utilise the same dialog used for creating the bots. The Dialog will have a "Create Mode" and an "Edit Mode", On the edit mode the title changes to "Edit <Bot Name> Bot " and the submit button changes to "Update Bot". The current bot details i.e name and description will also be autofilled on the form when the user selects the edit button on their bot menu.

The design for this feature is as shown below:

Screenshot 2022-10-18 at 08 46 05

The Dialog needs to appear and function as described above.

On the update dialog I can:
1. Edit my bot name.
2. Edit my bot description.

Your assignment is to reuse the current create dialog to support this feature.

### Technical Analysis

This feature can be found in libs/features/convs-mgr/home/modals/create-bot-modal

For this feature you'll have to:
```markdown

1. Create a function that returns a FormGroup patched will the current bot details.
2. Utilise Angular's ng-template to switch between texts on the FrontEnd based on the Form Mode.
3. Create a function that uses the story store to update the story.

```

### Acceptance Criteria

1. When rendered the component is pixel perfect compared to the design above.
2. The component is well structured following the Technical Analysis.
4. All functions of the form work seamlessly.

### PR DESCRIPTION
**Pull request by AnnaL001**
# Description

Adds a feature that allows for editing of a bot's details; name and description.  

The feature includes the following:
- [X] Edit bot form opens when the "Edit" bot menu actions item is selected. 
- [X] Edit bot form opens with the specified title and submit button text.
- [X] Edit bot form is populated with the current bot details when the dialog is opened.
- [X] Changes to bot details are saved when clicking the button with the text "Update Bot".

Fixes #103.

## Type of change
- [X] New feature (non-breaking change which adds functionality)

# Screenshot 
Screenshot from 2022-11-03 11-50-55

# Checklist:
- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: apps/conv-learning-manager/src/app/app.module.ts [hunk 1]
```diff
@@ -26,7 +26,7 @@ import { StoryBlocksStateModule } from '@app/state/convs-mgr/stories/blocks';
 import { StoryBlockConnectionsStateModule } from '@app/state/convs-mgr/stories/block-connections';
 
 
-import { environment } from '../environments/environments';
+import { environment } from '../environments/environment';
```
Comment (IanOdhiambo9) [inline apps/conv-learning-manager/src/app/app.module.ts:30]: Please gitignore your environments

**Review by IanOdhiambo9 — CHANGES_REQUESTED** [review]
Please find time to schedule a call with me to go over this together.

File: apps/conv-learning-manager/src/app/app.module.ts (hunk 1 above)
Comment (AnnaL001) [reply apps/conv-learning-manager/src/app/app.module.ts]: Change has been made

File: apps/conv-learning-manager/src/main.ts [hunk 2]
```diff
@@ -2,7 +2,7 @@ import { enableProdMode } from '@angular/core';
 import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
 
 import { AppModule } from './app/app.module';
-import { environment } from './environments/environments';
+import { environment } from './environments/environment';
```
[the remark this answers is not in the document]
Comment (AnnaL001) [reply apps/conv-learning-manager/src/main.ts]: Change has been made

**Review by IanOdhiambo9 — APPROVED** [review]
Good Job

**Pull request merged**